vendor/views/vendor.py

```diff
@@ -289,7 +281,7 @@ def get(self, request, *args, **kwargs):
 
     def post(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        invoice = get_purchase_invoice(request.user)
+        invoice = get_purchase_invoice(request.user, request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
invoice = get_purchase_invoice(request.user, get_site_from_request(request))